packages/actor-context-preprocess-statistic-link-dereference/lib/StatisticLinkDereference.ts

```diff
@@ -0,0 +1,27 @@
+import { StatisticBase } from '@comunica/bus-context-preprocess';
+import type { ILink, IQuerySource } from '@comunica/types';
+
+export class StatisticLinkDereference extends StatisticBase<ILink> {
+  public constructor() {
+    super();
+  }
+
+  public updateStatistic(link: ILink, source: IQuerySource): boolean {
+    const metadata: Record<string, any> = {
+      type: source.constructor.name,
+      dereferencedTimestamp: Date.now(),
```

Comment:
Open question: do we want low-resolution timestamps (like Date.now) or high-resolution timestamps (like Performance.now). The latter is more precise and more modern. https://developer.mozilla.org/en-US/docs/Web/API/Performance/now#performance.now_vs._date.now

Reply:
I wasn't aware of Performance.now(), given it's more accurate, using it is probably better.